src/scheduling/calculate_makespan.py

```diff
@@ -0,0 +1,132 @@
+"""__summary__"""
+from collections import defaultdict
+from copy import deepcopy
+
+import pulp
+
+from .types import JobResultInfo, LPInstance, PTimes, STimes
+
+
+def calculate_makespan(
+    lp_instance: LPInstance,
+    jobs: list[JobResultInfo],
+    process_times: PTimes,
+    setup_times: STimes,
+) -> float:
+    """Calculates the actual makespan from the list of results.
+
+    Executes the schedule with the corret p_ij and s_ij values.
+
+    Args:
+        lp_instance (LPInstance): The base LP instance.
+        jobs (list[JobResultInfo]): The list of job results.
+        process_times (PTimes): The correct  p_ij.
+        setup_times (STimes) The correct s_ij.
+
+    Returns:
+        float: The makespan of the schedule.
+    """
+    return _calc_makespan(
+        jobs,
+        process_times,
+        setup_times,
+        lp_instance.jobs,
+        lp_instance.machines,
+    )
+
+
+def _find_last_completed(
+    job_name: str, jobs: list[JobResultInfo], machine: str
+) -> JobResultInfo:
+    """Finds the last completed job before the given job from the original schedule."""
+    original_starttime = next(
+        (j.start_time for j in jobs if job_name == j.name),
+        0,
```

Comment:
It gives 0 when there is no job with a matching name, right? Why though?
Code should fail if something is not as intended.